website/static/js/profile.js

```diff
@@ -563,13 +572,82 @@ var SocialViewModel = function(urls, modes) {
         return false;
     });
 
+    self.addWebsiteInput = function() {
+        this.profileWebsites.push(ko.observable().extend({
+            ensureHttp: true
+        }));
+    };
+    
+    self.removeWebsite = function(profileWebsite) {
+        var profileWebsites = ko.toJS(self.profileWebsites()),
+            idx = profileWebsites.indexOf(profileWebsite);
+            bootbox.confirm({
+                title: 'Remove website?',
+                message: 'Are you sure you want to remove this website from your profile?',
+                callback: function(confirmed) {
+                    if (confirmed) {
+                        self.profileWebsites.splice(idx, 1);
+                        self.submit();
+                        self.changeMessage(
+                            'Website Removed',
+                            'text-danger',
+                            5000
+                        );
+                        if (self.profileWebsites().length === 0) {
+                            self.addWebsiteInput();
+                        }
+                    }
+                },
+                buttons:{
+                    confirm:{
+                        label:'Remove',
+                        className:'btn-danger'
+                    }
+                }
+            });
+    };
+
     self.fetch();
 };
 SocialViewModel.prototype = Object.create(BaseViewModel.prototype);
 $.extend(SocialViewModel.prototype, SerializeMixin.prototype, TrackedMixin.prototype);
 
-var ListViewModel = function(ContentModel, urls, modes) {
+SocialViewModel.prototype.serialize = function() {
+    var serializedData = ko.toJS(this);
+
+    function removeBlankValues(value) {
+        return value !== '';
+    }
+    
+    serializedData.profileWebsites = serializedData.profileWebsites.filter(removeBlankValues);
+    return serializedData;
+};
 
+SocialViewModel.prototype.unserialize = function(data) {
+    var self = this,
+        websiteValue = [];
+    $.each(data || {}, function(key, value) {
+        if (ko.isObservable(self[key]) && key === 'profileWebsites') {
+            if (value.length === 0) {
+                value.push('');
```

Comment:
I believe we tried this earlier, and the sequencing doesn't work because the information in the view will later be overwritten by the data in value.